pretrain/tinyllama.py

```diff
@@ -148,29 +148,29 @@ def train(fabric, state, train_dataloader, val_dataloader, resume):
     max_tokens_per_device = max_tokens // fabric.world_size
     tokens_per_iter = micro_batch_size * model.config.block_size
     max_iters = max_tokens_per_device // tokens_per_iter
+    initial_iter = state["iter_num"]
+    train_iterator = iter(train_dataloader)
+
+    # resume data loader state by fast-forwarding through all seen batches
+    # drop this once streaming dataset supports proper resuming
+    if resume:
+        resume_t0 = time.perf_counter()
+        for resume_iter in range(initial_iter + 1):
```

Comment:
I have doubts about this `+1` situation.
`state[inter_num]` is initialized with 0. So if the training loop is stopped right after the first step, the `inter_num` will be 1 and thus, during the resuming stage, we need to iterate over the `train_iterator` only once, which is exactly `range(initial_iter) -> range(1) --> [0]`.

Reply:
```suggestion
        for resume_iter in range(initial_iter):
```